gqlalchemy/loaders.py

```diff
@@ -0,0 +1,386 @@
+
+
+
+@dataclass(frozen=True)
+class Nx1Mapping:
+    mapping: RelationshipMapping
+    from_entity: bool = False
+
+
+
+@dataclass(frozen=True)
+class MxNMapping:
+    mapping1: RelationshipMapping
+    mapping2: RelationshipMapping
+    label: str
+    from_first: bool = True
+
+
+
+Mapping = Union[List[Nx1Mapping], MxNMapping]
+
+
+
+@dataclass
+class CollectionMapping:
+    collection_name: str
+    mapping: Mapping = None
+    indices: Optional[List[str]] = None
+
+
+
+class DataSource(ABC):
+    def __init__(self, file_extension: str) -> None:
+        self._file_extension: str = file_extension
+
+    
+    @abstractmethod
+    def load_data(
+        self,
+        collection_name: str,
+        is_cross_table: bool=False
+    ) -> None:
+        pass
+
+
+
+class S3DataSource(DataSource):
+    def __init__(
+        self, 
+        bucket_name: str,
+        s3_access_key: str,
+        s3_secret_key: str,
+        s3_region: str,
+        file_extension: str,
+        s3_session_token: Optional[str] = None,
+    ) -> None:
+        super().__init__(file_extension=file_extension)
+        self._bucket_name: str = bucket_name
+        self._s3_access_key: str = s3_access_key
+        self._s3_secret_key: str = s3_secret_key
+        self._s3_region: str = s3_region
+        self._s3_session_token: Optional[str] = s3_session_token
+
+    
+    def load_data(
+        self,
+        collection_name: str,
+        is_cross_table: bool=False,
+        columns: Optional[List[str]] = None,
+    ) -> None:
+        s3 = fs.S3FileSystem(
+            region=self._s3_region,
+            access_key=self._s3_access_key,
+            secret_key=self._s3_secret_key,
+            session_token=self._s3_session_token,
+        )
+        source = f"{self._bucket_name}/{collection_name}.{self._file_extension}"
+        print("Loading " + ("cross table " if is_cross_table else "") + f"data from {source}")
+
+        # Load dataset via Pyarrow
+        dataset = ds.dataset(source, filesystem=s3)
+
+        # Load batches from raw data
+        for batch in dataset.to_batches(
+            columns=columns,
+        ):
+            for batch in batch.to_pylist():
+                yield batch
+
+
+@dataclass(frozen=True)
+class Configuration:
+    label: Optional[str] = None
+    column_names_mapping: Optional[Dict[str, str]] = None
+
+
+class Translator:
+    _DIRECTION = {
+        True: ("a", "b"),
+        False: ("b", "a"),
+    }
+
+    def __init__(
+        self,
+        data_source: DataSource,
+        data_configuration: Dict[str, Any], 
+        memgraph: Optional[Memgraph] = None
+    ) -> None:
+        self._memgraph: Memgraph = memgraph if memgraph is not None else Memgraph()
+        self._data_source = data_source
+        self._collection_mappings: List[CollectionMapping] = []
+        self._cross_relationship_mappings: List[CollectionMapping] = []
+
+        self._load_name_mappings(data_configuration["name_mappings"])
```

Comment:
also rename this name_mappings

Reply:
This is not field name, it's method name